### DISCUSSION TITLE
sftp host_keys and host_certificates

### DISCUSSION THREAD

**Discussion by j-lanham** [drakkan/sftpgo#872] [General]
I just got through trying to setup sftpgo to use a wildcard certificate and key for the host certificate.  It's the same host certificate and private key I'm using for an existing ftps server that I want to use sftpgo to replace.  Unfortunately it won't accept this combination.  It keeps telling me it can't find the key even though I've provided it.

**Comment by drakkan** [comment]
Hi,

it seems you are confusing SFTP with FTPS, they are completely different protocols. SFTPGo supports both SFTP and FTPS. 
At least post your SFTPGo configuration so I can try to figure out what you are trying to achieve, thanks.

EDIT: take a look [here](https://github.com/drakkan/sftpgo/blob/main/docs/howto/lets-encrypt-certificate.md#enable-explicit-ftp-over-tls) to understand how to configure FTP over TLS: replace `/var/lib/sftpgo/certs/sftpgo.com.crt` and `/var/lib/sftpgo/certs/sftpgo.com.key` with the paths to your certificates, `host_certificates` and  `host_keys` are for SFTP and you can generally leave these settings empty

**Reply by j-lanham** [reply]
Actually, I'm not.  I'm just trying to provide a new capability that the other server lacked.  I have ftp setup in the configuration as ftps passive to take over for the existing server.  They should be able to share the certificates, should they not?

Also, is there a way to upload the configuration file privately or do I need to redact company information and upload it publicly?

And, I'm currently trying to run it under Windows, unfortunately.  Sor /var/lib, of course doesn't exist.

**Reply by drakkan** [reply]
> Actually, I'm not. I'm just trying to provide a new capability that the other server lacked. I have ftp setup in the configuration as ftps passive to take over for the existing server. They should be able to share the certificates, should they not?
> 

SFTPGo supports PEM encoded certificates (the most common format). If your certificate is PEM encoded it will work, otherwise you have to convert it

> Also, is there a way to upload the configuration file privately or do I need to redact company information and upload it publicly?
> 

No sorry, my private email is only for project [sponsors](https://github.com/sponsors/drakkan). I no longer reply to random people who write to my email address asking me for free support.

> And, I'm currently trying to run it under Windows, unfortunately. Sor /var/lib, of course doesn't exist.

This does not change anything, just set the Windows path to your certificate and remember to escape `\`, for example `C:\\data\\cert.crt`. The configuration file must be a valid json file
